### PR TITLE
[NFC][SPIRV] Fix for selectExtInst to be able to process intrinsics

### PR DESCRIPTION
**Pull request by farzonl**
`selectExtInst` tries to add the intrinsic to the SPIRV GL extension instruction.
`MO_IntrinsicID` is always the first operand when we come from `selectIntrinsic`. 
For all those cases `selectExtInst` needs to know to start at index 2.

### CODE REVIEW

**Review by bharadwajy — APPROVED** [review]
LGTM!

**Review by python3kgae — APPROVED** [review]

**Review by coopp — APPROVED** [review]
Looks good to me.  I didn't see any test changes in this PR, so I assume this change will be covered by existing tests?

**Comment by farzonl** [comment]
> Looks good to me. I didn't see any test changes in this PR, so I assume this change will be covered by existing tests?

Yeah nfc so existing tests not breaking for the intrinsic selects replaced is the test to show this works.

File: llvm/lib/Target/SPIRV/SPIRVInstructionSelector.cpp [hunk 1]
```diff
@@ -802,8 +784,12 @@ bool SPIRVInstructionSelector::selectExtInst(Register ResVReg,
                      .addImm(static_cast<uint32_t>(Set))
                      .addImm(Opcode);
       const unsigned NumOps = I.getNumOperands();
-      for (unsigned i = 1; i < NumOps; ++i)
-        MIB.add(I.getOperand(i));
+      unsigned Index = 1;
+      if (I.getOperand(Index).getType() ==
```
Comment (bogner) [inline llvm/lib/Target/SPIRV/SPIRVInstructionSelector.cpp:788]: Might be worth an assert that `NumOps > 2` so folks get a better/quicker error if they accidentally misuse this

**Review by bogner — APPROVED** [review]

File: llvm/lib/Target/SPIRV/SPIRVInstructionSelector.cpp (hunk 1 above)
Comment (bogner) [reply llvm/lib/Target/SPIRV/SPIRVInstructionSelector.cpp:788]: No, I think `NumOps > 1` is right - we're just making sure that the `I.getOperand(Index)` here (where `Index == 1`) will crash in a helpful way. This makes the assumption here more explicit.

Comment (farzonl) [reply llvm/lib/Target/SPIRV/SPIRVInstructionSelector.cpp:788]: Sorry misunderstood. I thought you wanted the assert in the if block. this makes more sense.

**Pull request merged**